aws-predictions-tensorflow/src/test/java/com/amplifyframework/predictions/tensorflow/service/TextClassificationTest.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.tensorflow.service;
+
+import com.amplifyframework.predictions.models.Sentiment;
+import com.amplifyframework.predictions.models.SentimentType;
+import com.amplifyframework.predictions.tensorflow.asset.TextClassificationDictionary;
+import com.amplifyframework.predictions.tensorflow.asset.TextClassificationLabels;
+import com.amplifyframework.predictions.tensorflow.asset.TextClassificationModel;
+import com.amplifyframework.testutils.random.RandomString;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test that sentiment detection using TensorFlow Lite interpreter
+ * to output Amplify text interpretation result works.
+ */
+public final class TextClassificationTest {
+
+    private TextClassificationModel mockInterpreter;
+    private TextClassificationDictionary mockDictionary;
+    private TextClassificationLabels mockLabels;
+
+    private TensorFlowTextClassificationService service;
+
+    /**
+     * Set up mock behavior before each test.
+     */
+    @Before
+    public void setUp() {
+        // Mock asset loaders
+        mockInterpreter = mock(TextClassificationModel.class);
+        mockDictionary = mock(TextClassificationDictionary.class);
```

Comment:
Instead of mocking the dictionary, can your provide `text_classification_vocab.txt`, and use real one?

Reply:
I was trying to stay away from relying on androidx and robolectric if possible. I think this is sufficient for now.

Reply:
There are pros and cons -- the less you mock, the more "real" your test is, and the more coverage you get. Plus, you'll be able to keep `final` on your class, to keep it "closed."

The only con is really just that Robolectric is slow.

Reply:
One more note -- 

I used to write (and I have seen a lot of others write) tests like this:

```
// Assuming a simple actor like ...
class ObjectUnderTest {
    private final Dependency dependency;
    ObjectUnderTest(Dependency dependency) {
        this.dependency = dependency;
    }
    Value lookupValue() {
        return dependency.getValue();
     }
}
```

```
// Here, we mock out all dependencies of the unit, and write a test
// that basically just re-codifies or implementation logic into Mockito syntax.
public final class AnemicUnitTest {
    private Dependency dependency;
    @Before
    public void setup() {
        this.dependency = mock(Dependency.class);
        this.objectUnderTest = new ObjectUnderTest(dependency);
    }

    @Test
    public void lookupValueCallsDependency() {
        when(dependency.getValue()).thenReturn(Value.of(2));
        assertEquals(Value.of(2), doSomething.lookupValue());
    }
}
```

This is an "anemic" unit test, where all dependencies are mocked. We've verified that the unit of code (the ObjectUnderTest) calls our Dependency. But so what? The complexity is very low.

Now, let's imagine `Dependency` is something like TensorFlow, and `getValue()` triggers _thousands_ of lines of code to be executed. If we can avoid using the mock, we have a bonafide component test on our hands, and a good indication that the `ObjectUnderTest` itself will work in a real-world application.

```
// Here, we test *both* ObjectUnderTest, and the direct dependencies it needs to its job
public final class ComponentTest {
     private RealDependency dependency;
     private ObjectUnderTest objectUnderTest;

     @Before
      public void setup() {
          this.dependency = new RealDependency();
          this.objectUnderTest = new ObjectUnderTest(dependency);
      }

      @Test
      public void lookupValueReturnsExpectedValue() {
          // Value is not arranged, but derived form real world business conditions
          dependency.setColor("#441155"); // Some logic mutates the dependency state
          assertTrue(Value.of(44), objectUnderTest.getValue());
      }
}
```